networkx/algorithms/shortest_paths/weighted.py

```diff
@@ -871,14 +879,23 @@ def _dijkstra_multisource(
             elif u not in seen or vu_dist < seen[u]:
                 seen[u] = vu_dist
                 push(fringe, (vu_dist, next(c), u))
-                if paths is not None:
-                    paths[u] = paths[v] + [u]
+                if paths_dict is not None:
+                    paths_dict[u] = paths_dict[v] + [u]
                 if pred is not None:
                     pred[u] = [v]
             elif vu_dist == seen[u]:
                 if pred is not None:
                     pred[u].append(v)
 
+    if target is not None and paths is not None:
+        # reconstruct the path from source to target
+        reversed_target_path = []
+        next_node = target
+        while next_node is not None:
+            reversed_target_path.append(next_node)
+            next_nodes = pred.get(next_node)
+            next_node = next_nodes and next_nodes[0]
+        paths[target] = list(reversed(reversed_target_path))
```

Comment:
If you use a `deque` you can `appendleft` and avoid having to reverse the list.
I also feel this is cleaner `get` logic, but you decide.
```suggestion
        next_node = target
        paths[target] = deque()
        while next_node is not None:
            paths[target].appendleft(next_node)
            next_node = pred.get(next_node, [None])[0]
        paths[target] = list(paths[target])
```

Reply:
I re-wrote this. I ended up using arraylist + in-place reversal. I believe this is still clear and should be faster. Did a quick experiment with 10k elements and list+in-place reverse is something like 40% faster. That is like ~2 ms for long lists such as 10k elements, but given that this method is one of the most used algorithms, I believe it's worth it.


```python

n = 10000

def list_append_reverse():
    path = []
    for i in range(n):
        path.append(i)
    path.reverse()
    return path

def deque_appendleft():
    path = deque()
    for i in range(n):
        path.appendleft(i)
    return list(path)

list_append_reverse_time = timeit.timeit(list_append_reverse, number=1000)
deque_appendleft_time = timeit.timeit(deque_appendleft, number=1000)

list_append_reverse_time, deque_appendleft_time
```